algorithms/src/snark/varuna/tests.rs

```diff
@@ -61,6 +65,17 @@ mod varuna {
                         let (index_pk, index_vk) = $snark_inst::circuit_setup(&universal_srs, &circ).unwrap();
                         println!("Called circuit setup");
 
+                        let max_domain_size = index_pk.circuit.max_domain_size();
+                        let max_degree = index_pk.circuit.max_degree();
+                        let coefficient_support = index_pk.circuit.index_info.get_degree_bounds::<Fr>();
```

Comment:
Yeah this seems to be a code smell; we shouldn't have to extract these details just to pass them in again.